segmentation/utils/spatialanalysis_utils_test.py

```diff
@@ -10,3 +15,232 @@ def test_calc_dist_matrix():
+    distmat[2:4, 1] = 50
+    distmat[1, 2:4] = 50
+    distmat[4:9, 1] = 150
+    distmat[1, 4:9] = 150
+    distmat[9, 1:9] = 200
+    distmat[1:9, 9] = 200
+    distmat[3, 2] = 50
+    distmat[2, 3] = 50
+    distmat[4:9, 2] = 150
+    distmat[2, 4:9] = 150
+    distmat[4:9, 3] = 150
+    distmat[3, 4:9] = 150
+    distmat[5:9, 4] = 50
+    distmat[4, 5:9] = 50
+    distmat[6:9, 5] = 50
+    distmat[5, 6:9] = 50
+    distmat[7:9, 6] = 50
+    distmat[6, 7:9] = 50
+    distmat[8, 7] = 50
+    distmat[7, 8] = 50
+
+    return cellarray, distmat
+
+
+def test_closenum():
+    # test the closenum function
+    test_cellarray, test_distmat = make_test_closenum()
+    test_thresholds = make_threshold_mat()
+
+    # subsets the expression matrix to only have marker columns
+    data_markers = test_cellarray.drop(test_cellarray.columns[[0, 1, 14, 23, 24, 25, 26, 27, 28, 29, 30]], axis=1)
+    # list of all markers
+    marker_titles = data_markers.columns
+    # length of marker list
+    marker_num = len(marker_titles)
+
+    # subsetting threshold matrix to only include column with threshold values
+    thresh_vec = test_thresholds.iloc[0:20, 1]
+
+    test_closenum, marker1_num, marker2_num = spatialorgmarkerexpression_utils.helper_function_closenum(
+        test_cellarray, data_markers, thresh_vec, test_distmat, marker_num, dist_lim=100, cell_label_idx=24)
+    assert (test_closenum[:2, :2] == 16).all()
+    assert (test_closenum[3:5, 3:5] == 25).all()
+    assert (test_closenum[5:7, 5:7] == 1).all()
+
+    # Now test with Erin's output
+    # cell_array = pd.read_csv(
+    #     "/Users/jaiveersingh/Downloads/SpatialEnrichm"
+    #     "ent/granA_cellpheno_CS-asinh-norm_revised.csv")
+    # marker_thresholds = pd.read_csv(
+    #     "/Users/jaiveersingh/Downloads/SpatialEnrichment/markerThresholds.csv")
+    # dist_matrix = np.asarray(pd.read_csv(
+    #     "/Users/jaiveersingh/Documents/MATLAB/distancesMat5.csv", header=None))
+    # closenum, closenumRand, z, muhat, sigmahat, p, h, adj_p, markertitles = \
+    #     spatialorgmarkerexpression_utils.calculate_channel_spatial_enrichment(
+    #         dist_matrix, marker_thresholds, cell_array)
+    # real_closenum = np.asarray(pd.read_csv(
+    #     "/Users/jaiveersingh/Documents/MATLAB/SpatialAnalysis/closeNum.csv"))
+    # assert np.array_equal(closenum, real_closenum)
+
+
+def test_closenumrand():
+    test_cellarray, test_distmat = make_test_closenum()
+
+    # generate random inputs to test shape
+    marker1_num = [random.randrange(0, 10) for i in range(20)]
+    marker2_num = [random.randrange(0, 5) for i in range(400)]
+    marker_num = 20
+
+    test_closenumrand = spatialorgmarkerexpression_utils.helper_function_closenumrand(
+        marker1_num, marker2_num, test_cellarray, test_distmat, marker_num, dist_lim=100,
+        cell_label_idx=24, bootstrap_num=100)
+
+    assert test_closenumrand.shape == (20, 20, 100)
+
+
+def test_calculate_enrichment_stats():
```

Comment:
This looks nice! Now let's do the same thing for the negative z score, and for non-significant.